fc_util/src/lib.rs

```diff
@@ -3,57 +3,189 @@
+impl fmt::Display for LocalTime {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}-{:02}-{:02}T{:02}:{:02}:{:02}.{:09}",
+            self.year + 1900,
+            self.mon,
+            self.mday,
+            self.hour,
+            self.min,
+            self.sec,
+            self.nsec
+        )
+    }
 }
 
-pub fn now_cputime_us() -> u64 {
+/// Returns a timestamp in nanoseconds from a monotonic clock.
+///
+/// Uses `_rdstc` on `x86_64` and [`get_time`](fn.get_time.html) on other architectures.
+pub fn timestamp_cycles() -> u64 {
+    #[cfg(target_arch = "x86_64")]
+    // Safe because there's nothing that can go wrong with this call.
+    unsafe {
+        std::arch::x86_64::_rdtsc() as u64
+    }
+    #[cfg(not(target_arch = "x86_64"))]
+    {
+        get_time(ClockType::Monotonic)
+    }
+}
+
+/// Returns a timestamp in nanoseconds based on the provided clock type.
+///
+/// # Arguments
+///
+/// * `clock_type` - Identifier of the Linux Kernel clock on which to act.
+pub fn get_time(clock_type: ClockType) -> u64 {
     let mut time_struct = libc::timespec {
         tv_sec: 0,
         tv_nsec: 0,
     };
     // Safe because the parameters are valid.
-    unsafe { libc::clock_gettime(libc::CLOCK_PROCESS_CPUTIME_ID, &mut time_struct) };
-    timespec_to_us(&time_struct)
+    unsafe { libc::clock_gettime(clock_type.into(), &mut time_struct) };
+    (time_struct.tv_sec as u64) * SECONDS_TO_NANOS + (time_struct.tv_nsec as u64)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::cmp::Ordering;
 
     #[test]
-    fn test_timestamp_cycles() {
+    fn test_get_time() {
         for _ in 0..1000 {
-            assert!(timestamp_cycles() < timestamp_cycles());
+            assert!(get_time(ClockType::Monotonic) <= get_time(ClockType::Monotonic));
         }
-    }
 
-    #[test]
-    fn test_now_cputime_us() {
         for _ in 0..1000 {
-            assert!(now_cputime_us() <= now_cputime_us());
+            assert!(get_time(ClockType::ProcessCpu) <= get_time(ClockType::ProcessCpu));
         }
+
+        for _ in 0..1000 {
+            assert!(get_time(ClockType::ThreadCpu) <= get_time(ClockType::ThreadCpu));
+        }
+
+        assert_ne!(get_time(ClockType::Real), 0);
+    }
+
+    #[test]
+    fn test_local_time_display() {
+        let local_time = LocalTime {
+            sec: 30,
+            min: 15,
+            hour: 10,
+            mday: 4,
+            mon: 7,
+            year: 119,
+            nsec: 123_456_789,
+        };
+        assert_eq!(
+            String::from("2019-07-04T10:15:30.123456789").cmp(&local_time.to_string()),
+            Ordering::Equal
+        );
+
+        let local_time = LocalTime {
+            sec: 5,
+            min: 5,
+            hour: 5,
+            mday: 23,
+            mon: 8,
+            year: 44,
+            nsec: 123,
+        };
+        assert_eq!(
```

Comment:
Why not use `assert_eq!(String::from("1944-08-23T05:05:05.000000123"), local_time.to_string());`